test-utils-core/src/main/java/uk/gov/justice/services/test/utils/core/schema/SchemaComplianceChecker.java

```diff
@@ -0,0 +1,371 @@
+package uk.gov.justice.services.test.utils.core.schema;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.counting;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.toList;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+
+import difflib.Delta;
+import difflib.DiffUtils;
+import difflib.Patch;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SchemaComplianceChecker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SchemaComplianceChecker.class);
+
+    private static final String WINDOWS_NEWLINE = "\r\n";
```

Comment:
Should be able to use system property to get path character?

Reply:
I cannot guarantee that the system delimiter is going to be the same as the one used in the file, so therefore i would say no.

Reply:
Use a Reader, you don't have to worry about the line endings.